test/unit/math/rev/fun/LDLT_factor_test.cpp

```diff
@@ -86,93 +47,20 @@ TEST(AgradRevMatrix, solve) {
   stan::math::recover_memory();
 }
 
-TEST(AgradRevMatrix, vectorD) {
-  using stan::math::LDLT_factor;
-  using stan::math::var;
-
-  Eigen::Matrix<var, -1, -1> A(2, 2);
-  A << 2, 1, 1, 2;
-  Eigen::Matrix<double, -1, -1> vectorD(2, 1);
-
-  Eigen::Matrix<double, -1, -1> A_double(2, 2);
-  A_double << 2, 1, 1, 2;
-  Eigen::LDLT<Eigen::Matrix<double, 2, 2> > ldlt_double(A_double);
-  Eigen::Matrix<double, -1, -1> expected_vectorD(2, 1);
-  expected_vectorD = ldlt_double.vectorD();
-
-  LDLT_factor<var, -1, -1> ldlt_A(A);
-  ASSERT_TRUE(ldlt_A.success());
-  EXPECT_NO_THROW(vectorD = ldlt_A.vectorD());
-  for (int i = 0; i < 2; i++)
-    EXPECT_FLOAT_EQ(expected_vectorD(i), vectorD(i));
-
-  stan::math::recover_memory();
-}
-
-TEST(AgradRevMatrix, rows) {
-  using stan::math::LDLT_factor;
-  using stan::math::var;
-
-  Eigen::Matrix<var, -1, -1> A(2, 2);
-  A << 2, 1, 1, 2;
-
-  LDLT_factor<var, -1, -1> ldlt_A(A);
-  ASSERT_TRUE(ldlt_A.success());
-  EXPECT_EQ(2U, ldlt_A.rows());
-
-  stan::math::recover_memory();
-}
-
-TEST(AgradRevMatrix, cols) {
+TEST(AgradRevMatrix, matrix) {
   using stan::math::LDLT_factor;
   using stan::math::var;
 
   Eigen::Matrix<var, -1, -1> A(2, 2);
   A << 2, 1, 1, 2;
+  auto ldlt_A = stan::math::make_ldlt_factor(A);
 
-  LDLT_factor<var, -1, -1> ldlt_A(A);
-  ASSERT_TRUE(ldlt_A.success());
-  EXPECT_EQ(2U, ldlt_A.cols());
-
-  stan::math::recover_memory();
-}
-
-TEST(AgradRevMatrix, compute) {
-  using stan::math::LDLT_factor;
-  using stan::math::var;
-
-  Eigen::Matrix<var, -1, -1> A(2, 2);
-  A << 2, 1, 1, 2;
   Eigen::Matrix<double, -1, -1> A_double(2, 2);
   A_double << 2, 1, 1, 2;
+  auto ldlt_double = stan::math::make_ldlt_factor(A_double);
 
-  Eigen::LDLT<Eigen::Matrix<double, -1, -1> > ldlt_double(A_double);
-  Eigen::Matrix<double, -1, -1> expected_mat, mat;
-
-  LDLT_factor<var, -1, -1> ldlt_A;
-
-  // tests on A: [2, 1][1, 2]
-  // only way to test is through side-effects.
-  EXPECT_NO_THROW(ldlt_A.compute(A));
-  ASSERT_TRUE(ldlt_A.success());
-
-  EXPECT_NO_THROW(mat = ldlt_A.alloc_->ldlt_.matrixLDLT());
-  expected_mat = ldlt_double.matrixLDLT();
-  for (int i = 0; i < 2; i++)
-    for (int j = 0; j < 2; j++)
-      EXPECT_FLOAT_EQ(expected_mat(i, j), mat(i, j))
-          << "element (" << i << ", " << j << ")";
-
-  // tests on A: [0, 0][0, 0]
-  A << 0, 0, 0, 0;
-  EXPECT_NO_THROW(ldlt_A.compute(A));
-  ASSERT_FALSE(ldlt_A.success());
-
-  // tests on A: [1, 2, 3][2, 3, 4]
-  A.resize(2, 3);
-  A << 1, 2, 2, 3, 3, 4;
-  EXPECT_THROW(ldlt_A.compute(A), std::invalid_argument);
-  ASSERT_FALSE(ldlt_A.success());
+  // EXPECT_MATRIX_EQ(A_double, ldlt_double.matrix());
```

Comment:
commented code?

Reply:
Good news! The `.matrix()` function to the prim version of `LDLT_factor` was broken. Uncommenting this revealed a bug. So it is fixed! So your efforts reviewing the giant pull were not for naught.